docs/source/techniques/SANS-Toml-v1.rst

```diff
@@ -1284,36 +2068,67 @@ set centre a b
 
 ..  code-block:: none
 
-    set centre 84.2 -196.5
+    SET CENTRE 84.2 -196.5
 
-**Existing Replacement**
+**Replacement Example**
 
 ..  code-block:: none
 
     [detector]
       [detector.configuration]
         # This will set both front and rear to the same centre values.
-        all_centre = {x=a, y=b}
+        all_centre = {x=0.0842, y=-0.1965}
 
-
-set centre a b c d [/MAIN] [/HAB]
+SET CENTRE[/MAIN][/HAB] a b [c d]
 ---------------------------------
 
+This command was used to specify the (x,y) coordinates (in real-space) of
+the centre of the scattering pattern on a specific detector. Compare with
+SET CENTRE a b.
+
+If the /MAIN qualifier was omitted the command was assumed to apply to
+the main (ie, rear) detector anyway. The /HAB qualifier was required to
+specify the beam centre coordinates for a high-angle (ie, front) detector.
+
+The parameters c and d allowed the size of the detector pixels in x & y
+to be passed to the data reduction.
+
+Approximate centre coordinates on the ISIS SANS instruments (which should
+be optimised using the beam centre finder tool!) are:
+
+..  code-block:: none
+
+    LARMOR: ( 0.020,  1.000)
+    LOQ:    ( 0.320,  0.320)
+    SANS2D: ( 0.100, -0.080)
+    ZOOM:   (-0.170, -0.050)
+
+**Note: In 2020 it was discovered that due to a forever bug in the legacy
```

Comment:
So this command is parsed correctly by both the legacy and existing parser. It was only `SET SCALES` which had the problems...

Reply:
Ok, I have a couple of questions here:

1. So you're saying if the legacy parser encountered SET CENTRE a b c d it used (a,b) as the centre coords and passed pixel sizes c & d to the data reduction, right? What did it do if c & d were _not_ specified?

2. When you say this command is also parsed by the existing parser, do you mean the TOML parser? If so, what is the replacement syntax for specifying c & d?

Reply:
> 1. So you're saying if the legacy parser encountered SET CENTRE a b c d it used (a,b) as the centre coords and passed pixel sizes c & d to the data reduction, right? What did it do if c & d were not specified?

It would throw `SetParser: Expected 5 entries for the SCALES setting, but got n` (see next link)

> 2. When you say this command is also parsed by the existing parser, do you mean the TOML parser? If so, what is the replacement syntax for specifying c & d?

Under the hood `SET SCALES` parses into `s, a, b, c, d`:
https://github.com/mantidproject/mantid/blob/b4bb8da80b0e8824dee5cec948f01f22ea948d08/scripts/SANS/sans/user_file/user_file_parser.py#L1242

However only `s` was taken (and we only have space for s in the data structure) here (notice it `scales.s`):
https://github.com/mantidproject/mantid/blob/b4bb8da80b0e8824dee5cec948f01f22ea948d08/scripts/SANS/sans/user_file/txt_parsers/ParsedDictConverter.py#L931

And the state object only has space for `s`:
https://github.com/mantidproject/mantid/blob/b4bb8da80b0e8824dee5cec948f01f22ea948d08/scripts/SANS/sans/state/StateObjects/StateScale.py#L30

Unfortunately, this pre-dates me so I can't say why it's doing what it's doing or what the correct behaviour is. So there's still no support for the 4 other values

Reply:
One of us is confused here, @DavidFair ! My questions above were about SET CENTRE, but you've replied about SET SCALES!

Reply:
https://github.com/mantidproject/mantid/blob/main/scripts/SANS/sans/user_file/user_file_parser.py#L1183

```
        An undocumented feature is:
        SET CENTRE[/MAIN|/HAB] x y [d1 d2]
        where d1 and d2 are pixel sizes. This is not used in the old parser, but user files have it nonetheless.
```

You'll notice at https://github.com/mantidproject/mantid/blob/main/scripts/SANS/sans/user_file/user_file_parser.py#L1250 and it simply extracts two float values, so `d1` and `d2` (from the above docs) are ignored AFAIK

This is probably why I ended up looking at SCALES by accident instead

Reply:
So how is the size of the detector pixels handled? What values does Mantid use?